rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -39,6 +39,25 @@
 
 namespace YAML
 {
+/// Pass metadata version to the sub-structs of BagMetadata for deserializing.
+/**
+  * Encoding should always use the current metadata version, so it does not need this value.
+  * We cannot extend the YAML::Node class to include this, so we must call it
+  * as a function with the node as an argument.
+  */
+template<typename T>
+T decode_for_version(const Node & node, int version)
+{
+  if (!node.IsDefined()) {
+    throw TypedBadConversion<T>(node.Mark());
+  }
+  T value;
```

Comment:
can this be default initialized? `T value{}`?

Reply:
yeah